demos/inpage/src/EthereumApi.tsx

```diff
@@ -327,12 +335,16 @@ export default class EthereumApi {
         callGasLimit: actions.map((a) => BigInt(a.gas)).reduce((a, b) => a + b),
         verificationGasLimit: temporaryEstimationGas,
         preVerificationGas: temporaryEstimationGas,
-        maxFeePerGas: feeData.maxFeePerGas,
-        maxPriorityFeePerGas: feeData.maxPriorityFeePerGas,
+        maxFeePerGas,
+        maxPriorityFeePerGas,
         paymasterAndData: '0x',
         signature: temporarySignature,
       } satisfies UserOperationStruct;
 
+      if (this.#waxInPage.getConfig('useTopLevelCompression')) {
+        userOp.callGasLimit = roundUpPseudoFloat(userOp.callGasLimit);
+      }
```

Comment:
Given significant portions of this code rely on `useTopLevelCompression` flag, could extract to separate function.